libkbfs/tlf_journal.go

```diff
@@ -402,33 +416,134 @@ func (j *tlfJournal) flush(ctx context.Context) (err error) {
 }
 
 func (j *tlfJournal) flushOneBlockOp(ctx context.Context) (bool, error) {
-	j.lock.Lock()
-	defer j.lock.Unlock()
-	return j.blockJournal.flushOne(ctx, j.delegateBlockServer, j.tlfID)
+	j.flushLock.Lock()
+	defer j.flushLock.Unlock()
+
+	o, e, data, serverHalf, err := func() (
```

Comment:
I think it would be cleaner in this instance to do:
```go
j.journalLock.RLock()
o, e, data, serverHalf, err := j.blockJournal.getNextEntryToFlush(ctx)
j.journalLock.RUnlock()
```

You lose the defense against panics, but a panic is application-breaking anyway.